### ISSUE TITLE
Backend Users

### ISSUE THREAD

**Issue by JonKrone** [travel-consensus/travel-consensus#78]
@prozack 

We haven't touched trying to implement users on the backend yet if you want to give it a shot.
What would we need to do to make it happen?

**Comment by prozack** [comment]
@JonKrone 
Other than signin/signup, building out authentication and the password hash?  We could have the invite user first check to see if the email address is in the user db, and then automatically build out the link in trip_users.  If a match is not found, it could enter some info into the user field to 'seed' the entry pending signup.  Like put a prelim user in with email and id, and link it on the trip_users db.

**Comment by JonKrone** [comment]
Yeah, search users by email and add them to the trip_users table if a user exists. seeding the database if the email doesn't exist is probably more complicated than we need now, though. On signup, check if user information exists, redirect to signin if it does and create account if it doesn't. on signin, check if email+password are correct then send a cookie if so. We can then start requiring signed-in users to access trip pages (only the ones they've been invited to)

**Comment by prozack** [comment]
@JonKrone 
I started looking into Passport to handle authentication and persistent session.  http://passportjs.org/
This seems like a pretty simple answer to what we want to do.  Specifically the passport-local module.  We can also implement the shortly express method with same level of fuss, but we may need to set up a session db in schema.  I don't want to go too far down rabbit hole before discussing further.

**Comment by JonKrone** [comment]
Looking at passport, it doesn't look too bad to work with and implement. My only concern is that I've heard auth services can be a real pain to configure and debug and get working properly but I'm sure its doable - just read the docs thoroughly. I'm taking the day at least half off so take a stab or two and let me know how it goes. I'll be available on slack.
